src/library/library.cpp

```diff
@@ -422,6 +422,13 @@ void Library::slotSetTrackTableRowHeight(int rowHeight) {
     emit(setTrackTableRowHeight(rowHeight));
 }
 
-void Library::onEvictingTrackFromCache(GlobalTrackCacheLocker* pCacheLocker, Track* pTrack) {
+void Library::afterEvictedTrackFromCache(GlobalTrackCacheLocker* pCacheLocker, Track* pTrack) {
+    // The evictor callback should always be called from the main
+    // thread, because the TrackDAO with the database connection
+    // is still bound to the main thread! We have to rethink this
+    // assertion after the multi-threaded database access layer is
+    // ready!
+    // TODO: Finish multi-threaded database access layer
+    DEBUG_ASSERT(QApplication::instance()->thread() == QThread::currentThread());
```

Comment:
How is that guranteed? I can easily imagine a situation where a worker thread executed the deleter.
Can't we use here the same trick delete later does?

Reply:
We cannot guarantee it, at least not now. But instead of closing our eyes and hoping for the best I added some debug assertions to spot those dangerous invocations if they occur. It's not a fix, just a safety guard that should alert us whenever we leave the safe ground of the current design.

I hope you understand that I'm not able to fix the multitude of unsafe legacy code in Mixxx all at once and all on myself.

Reply:
OK.